### DISCUSSION TITLE
How to use value files from remote charts

### DISCUSSION THREAD

**Discussion by javadev11** [helmfile/helmfile#1670] [Questions]
Hi,
I have charts that stored in our Helm repository (Artifactory) that have already customized values per each environment, e.g. "values.dev.yaml", "values.sit.yaml", "values.uat.yaml", "values.prod.yaml", and I need to use it when deploying charts to corresponding environment.  

**Is there any notation in helmfile values section to use chart's values files that stored in the chart's tgz package ?**
I'm aware of https://helmfile.readthedocs.io/en/latest/#loading-remote-environment-values-files where it can load values from S3 or http but it does not specify how to access "current" remote chart's values and most relevant that I find is :

```
staging:
     values:
      - http://$HOSTNAME/artifactory/example-repo-local/test.tgz@values.yaml #Artifactory url
```

To sum up what I'm looking for is something like below (assuming that I'm deploying chart to **dev** environment):

```
repositories:
 - name: artifactory
   url: https://artifactory.corp.net

releases:
- name: mychart
  chart:  artifactory/mychart
  ....
  values:
   -  artifactory/mychart@values.dev.yaml
```
P.S. I can try to use "http" go getter way, but then I need to specify full http path including version of each chart which is doable but not nice considering that I just told helmfile where my remote chart located and which version to use and need to repeat the same metadata in the values section.

**Comment by cloudchristina** [comment]
Hi @javadev11 are you able to try this?
```
environments:
  dev:
    values:
      - values.dev.yaml

  sit:
    values:
      - values.sit.yaml

  uat:
    values:
      - values.uat.yaml

  prod:
    values:
      - values.prod.yaml

releases:
  - name: your-release-name
    namespace: your-namespace
    chart: path/to/your/chart
    values:
      - {{ .Environment.Values | first }}
```

When you test from local, try `helmfile -e dev apply`

**Reply by javadev11** [reply]
hi @cloudchristina ,
thank you for the suggestion but imy question is about values from **remote** charts, not local
